src/SingleProject/Resizetizer/src/GeneratePackageAppxManifest.cs

```diff
@@ -0,0 +1,345 @@
+		//private static readonly XNamespace xmlns = "http://schemas.microsoft.com/appx/manifest/foundation/windows10";
+
+		private static readonly XNamespace xmlnsUap = "http://schemas.microsoft.com/appx/manifest/uap/windows10";
+
+		[Required]
+		public string IntermediateOutputPath { get; set; } = null!;
+
+		[Required]
+		public ITaskItem AppxManifest { get; set; } = null!;
+
+		public string? GeneratedFilename { get; set; }
+
+		public string? ApplicationId { get; set; }
+
+		public string? ApplicationDisplayVersion { get; set; }
+
+		public string? ApplicationVersion { get; set; }
+
+		public string? ApplicationTitle { get; set; }
+
+		public ITaskItem[]? AppIcon { get; set; }
+
+		public ITaskItem[]? SplashScreen { get; set; }
+
+		[Output]
+		public ITaskItem GeneratedAppxManifest { get; set; } = null!;
+
+		public override bool Execute()
+		{
+			try
+			{
+				Directory.CreateDirectory(IntermediateOutputPath);
+
+				var filename = Path.Combine(IntermediateOutputPath, GeneratedFilename ?? "Package.appxmanifest");
+
+				var appx = XDocument.Load(AppxManifest.ItemSpec);
+
+				UpdateManifest(appx);
+
+				appx.Save(filename);
+
+				GeneratedAppxManifest = new TaskItem(filename);
+			}
+			catch (Exception ex)
+			{
+				Log.LogErrorFromException(ex);
+			}
+
+			return !Log.HasLoggedErrors;
+		}
+
+		void UpdateManifest(XDocument appx)
+		{
+			var appIconInfo = AppIcon?.Length > 0 ? ResizeImageInfo.Parse(AppIcon[0]) : null;
+			var splashInfo = SplashScreen?.Length > 0 ? ResizeImageInfo.Parse(SplashScreen[0]) : null;
+			var imageExtension = ".png";
+
+			var xmlns = appx.Root!.GetDefaultNamespace();
+
+			// <Identity Name="" Version="" />
+			if (!string.IsNullOrEmpty(ApplicationId) || !string.IsNullOrEmpty(ApplicationDisplayVersion) || !string.IsNullOrEmpty(ApplicationVersion))
+			{
+				// <Identity>
+				var xidentity = xmlns + "Identity";
+				var identity = appx.Root.Element(xidentity);
+				if (identity == null)
+				{
+					identity = new XElement(xidentity);
+					appx.Root.Add(identity);
+				}
+
+				// Name=""
+				if (!string.IsNullOrEmpty(ApplicationId))
+				{
+					Guid.Parse(ApplicationId);
+
+					var xname = "Name";
+					var attr = identity.Attribute(xname);
+					if (attr == null || string.IsNullOrEmpty(attr.Value))
+						identity.SetAttributeValue(xname, ApplicationId);
+				}
+
+				// Version=""
+				if (!string.IsNullOrEmpty(ApplicationDisplayVersion) || !string.IsNullOrEmpty(ApplicationVersion))
+				{
+					var parts = ApplicationDisplayVersion?.Split('.') ?? Array.Empty<string>();
+					var v = new int[4];
+					for (var i = 0; i < 4 && i < parts.Length; i++)
+						v[i] = int.Parse(parts[i]);
+					if (!string.IsNullOrEmpty(ApplicationVersion))
+						v[3] = int.Parse(ApplicationVersion);
+
+					var xname = "Version";
+					var attr = identity.Attribute(xname);
+					if (attr == null || string.IsNullOrEmpty(attr.Value))
+						identity.SetAttributeValue(xname, $"{v[0]:0}.{v[1]:0}.{v[2]:0}.{v[3]:0}");
+				}
```

Comment:
Is the `Version` here required to be four parts on Windows?

If that is the case, we should just require `$(ApplicationDisplayVersion)` to be four parts on Windows. Then all our templates can just put `1.0.0.0` in there.

Reply:
It is required to be 4 parts on Windows as it is the normal assembly versioning format. However, I saw text in the macos docs saying this needs to be 3 parts for ios/maccat.

And, we need a way to get the version number/code in there as well. So it is a merge of 2 props that the other plats have separate places.

Reply:
Is it the right thing to merge them? We've not done this anywhere else or mentioned it in the spec:

https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/OneDotNetSingleProject.md

What if you just append `.0.0` where needed at the end?

Reply:
But then what do we do with the ApplicationVersion? Unlike android/ios which uses the number as the real version and the display is UI, Windows has one which is both. That is my real issue here. Unless we do something entirely different...

Reply:
But then this might cause an issue if the user keeps using say `1.0` as the ApplicationDisplayVersion, but increments the ApplicationVersion. For iOS/Android, the ApplicationVersion is the real version, but in Windows, the ApplicationDisplayVersion now becomes the main version.

And the word "Display" in ApplicationDisplayVersion is not really what the version is doing. It is the "actual" version now.

I suppose it is more the difference in behaviour.

In addition, we have APIs in essentials that get this "version code", so we need a way to represent this in the csproj and then a way to retrieve it from the package manifest. I was leveraging the 4 digit since  the display version can only be a max of 3 digits - at least that is what the ios docs say.

Reply:
`$(ApplicationDisplayVersion)` is `versionName` on Android, and so you can actually put anything there.

Maybe combining the values is fine, I would just add some test cases like I did here:

https://github.com/xamarin/xamarin-android/blob/33a6d1e52b7dc5548d425592a71372c85e43e8a3/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/SingleProjectTest.cs#L17-L36

You might have a test with `ApplicationVersion=1` and `ApplicationDisplayVersion=1.0.0.0` to make sure it at least gives a decent error message.

Reply:
I added some test cases https://github.com/dotnet/maui/pull/2704/files#diff-72359a1a0d7753fa498d335e15092b820f7034229d435acace0c8706707ee158R114-R136

Basically the display version is a 3 part semver and the version is an integer - like Android. You can technically use a 4 part display version if need be, but then you also need to make sure the version number is empty.